setup.py

```diff
@@ -28,6 +28,7 @@ def main():
                 "mock",
                 "nose",
                 "requests",
+                "numpy",
```

Comment:
@nicoddemus 
Adding numpy to all builds make the numpy tox env obsolete, doesn't it?
I'd rather not have it installed always, but in case this is kept it should clean up the env, jobs, and conditional skips we have for it.

Reply:
Oh right, thanks for catching that.

Practically I think it makes sense to install an additional dependency in order to have less jobs, as an extra dependency is usually much quicker to install than starting up an entire new job, which potentionally might run all tests again.

I will follow up with a fix.

Reply:
Just started with it already myself.

Reply:
I wanted to keep numpy optional, but what you've said sounds different?
Keep in mind that numpy might be slow to install without a wheel.

Reply:
You are right.

Let's remove `numpy` from `extras` and use `importorskip` on the test.

I have the diff ready:

```diff
diff --git a/setup.py b/setup.py
index be38857b0..dcf63f6fd 100644
--- a/setup.py
+++ b/setup.py
@@ -28,7 +28,6 @@ def main():
                 "mock",
                 "nose",
                 "requests",
-                "numpy",
                 "xmlschema",
             ]
         },
diff --git a/testing/python/fixtures.py b/testing/python/fixtures.py
index 4fcab0245..9ef7f0245 100644
--- a/testing/python/fixtures.py
+++ b/testing/python/fixtures.py
@@ -4190,6 +4190,7 @@ def test_indirect_fixture_does_not_break_scope(testdir):


 def test_fixture_parametrization_nparray(testdir):
+    pytest.importorskip("numpy")
     testdir.makepyfile(
         """
         from numpy import linspace
```